wagtail/documents/views/documents.py

```diff
@@ -22,74 +23,83 @@
 permission_checker = PermissionPolicyChecker(permission_policy)
 
 
-@permission_checker.require_any('add', 'change', 'delete')
-@vary_on_headers('X-Requested-With')
-def index(request):
-    Document = get_document_model()
+class BaseListingView(TemplateView):
+    @method_decorator(permission_checker.require_any('add', 'change', 'delete'))
+    def get(self, request):
```

Comment:
question: given all this does is call `TemplateView`'s counterparts, is it needed?